lib/builder.js

```diff
@@ -219,6 +220,207 @@ Builder.prototype.traceModule = function(moduleName) {
   });
 };
 
+/**
+ * Optimizes a set of entry points
+ * @param {String[]|Object} entryPoints Array of entry points or Object of entry points with entry point names as keys. Entry points are Strings.
+ * @param {Function({String[]|Object} entryPoints, {Object} trace, {Object} optimizationOptions) => {{String} name, {String[]|Object} entryPoints, {Object[]} modules}[]} optimizationFunction A function to perform the optimization
+ * @param  {{String: [outPath], Boolean: sourceMaps=true, Boolean: minify=true, Boolean: uglify=true}} opts Options defining how optimization works. Additional parameters specific to optimizationFunction can be added to this object.
+ * @return {Thenable} Promise resolved with an Array of bundles
+ * @example <caption>Example use of builder.optimizeBuild</caption>
+ * // For more information, especially on the optimizationFunction, please see the readme docs.
+ * builder.optimizeBuild({
+ *   // entryPoints can also be an Array if you don't care about the names of the bundles
+ *   other: ['src/stuff', 'src/otherStuff'],
+ *   help: 'src/help',
+ *   main: ['src/main', 'src/secondary', 'src/tertiary']
+ * },
+ * require('custom-optimizer'),
+ * {
+ *   outPath: 'out/folder', // optional, if not set returns source as memory compilation
+ *   sourceMaps: true, // optional, defaults to true
+ *   minify: true, // optional, defaults to true
+ *   uglify: true, // optional, defaults to true
+ *   // this option is specific to the optimisation function and may vary between implementations
+ *   entryPointPriorities: {
+ *     main: 0,
+ *     help: 1,
+ *     other: 2
+ *   },
+ * });
+ */
+Builder.prototype.optimizeBuild = function(entryPoints, optimizationFunction, opts) {
```

Comment:
No problem... builder.js reference it and include it in the prototype or should optimizer be completely independent from the Builder object?